### PR TITLE
Split Web & Mobile screens as they vary too much

### PR DESCRIPTION
**Pull request by t3chguy**

### CODE REVIEW

**Review by gsouquet — APPROVED** [review]

File: types/kotlin2/MobileScreen.kt [hunk 1]
```diff
@@ -16,23 +16,28 @@
 
 package im.vector.app.features.analytics.plan
 
-import im.vector.app.features.analytics.itf.VectorAnalyticsScreen
+import im.vector.app.features.analytics.itf.VectorAnalyticsEvent
```
Comment (pixlwave) [inline types/kotlin2/MobileScreen.kt:19]: The python generator needs to be updated to fix this.

File: types/swift/MobileScreen.swift [hunk 2]
```diff
@@ -19,9 +19,10 @@ import Foundation
 // GENERATED FILE, DO NOT EDIT. FOR MORE INFORMATION VISIT
 // https://github.com/matrix-org/matrix-analytics-events/
 
-/// Triggered when the user changed screen
+/// Triggered when the user changed screen on Element Android/iOS
 extension AnalyticsEvent {
-    public struct Screen: AnalyticsScreenProtocol {
+    public struct MobileScreen: AnalyticsEventProtocol {
```
Comment (pixlwave) [inline types/swift/MobileScreen.swift:24]: Same for this, it should conform to `AnalyticsScreenProtocol`

File: types/kotlin2/MobileScreen.kt [hunk 3]
```diff
@@ -16,23 +16,28 @@
...
  */
-data class Screen(
+data class MobileScreen(
         /**
          * How long the screen was displayed for in milliseconds.
          */
         val durationMs: Int? = null,
         val screenName: ScreenName,
-) : VectorAnalyticsScreen {
+) : VectorAnalyticsEvent {
```
Comment (pixlwave) [inline types/kotlin2/MobileScreen.kt:33]: And this.

**Review by pixlwave — COMMENTED** [review]
Just a small comment on the python generator.

File: types/kotlin2/MobileScreen.kt (hunk 1 above)
Comment (t3chguy) [reply types/kotlin2/MobileScreen.kt:19]: Good spot, thanks

**Review by pixlwave — APPROVED** [review]

File: types/kotlin2/WebScreen.kt [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+import im.vector.app.features.analytics.itf.VectorAnalyticsEvent
+
+// GENERATED FILE, DO NOT EDIT. FOR MORE INFORMATION VISIT
+// https://github.com/matrix-org/matrix-analytics-events/
+
+/**
+ * Triggered when the user changed screen on Element Web/Desktop
+ */
+data class WebScreen(
+        val $current_url: $current_url,
```
Comment (bmarty) [inline types/kotlin2/WebScreen.kt:28]: This is not valid Kotlin code.
I will just (temporarily) delete this file on our side (in https://github.com/vector-im/element-android/pull/5223), but we should definitely validate the code from here by trying to build the generated file.
Why is the format different than the for the other files?

File: types/kotlin2/WebScreen.kt [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+        /**
+         * How long the screen took to load, if applicable.
+         */
+        val durationMs: Int? = null,
+) : VectorAnalyticsEvent {
+
+    enum class $current_url {
```
Comment (bmarty) [inline types/kotlin2/WebScreen.kt:35]: This line does not compile too.

File: types/swift/WebScreen.swift [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+// GENERATED FILE, DO NOT EDIT. FOR MORE INFORMATION VISIT
+// https://github.com/matrix-org/matrix-analytics-events/
+
+/// Triggered when the user changed screen on Element Web/Desktop
+extension AnalyticsEvent {
+    public struct WebScreen: AnalyticsEventProtocol {
+        public let eventName = "$pageview"
+
+        public let $current_url: $current_url
```
Comment (bmarty) [inline types/swift/WebScreen.swift:27]: I am pretty sure this is not valid Swift as well.

**Review by bmarty — COMMENTED** [review]
Problem with generated code :/

**Comment by t3chguy** [comment]
We should have a way for Web* to not build into Kt/Swift, and for Mobile* stuff to not build into TS

File: types/swift/WebScreen.swift (hunk 6 above)
Comment (pixlwave) [reply types/swift/WebScreen.swift:27]: You are right, I missed this when I reviewed on GH
> the '$' prefix is reserved for implicitly-synthesized declarations

**Pull request merged**